tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -33,3 +33,17 @@ def unrecognized_help_parameter_rule(linter, help_entry):
             violations.append(param_help_name)
     if violations:
         raise RuleError('The following parameter help names are invalid: {}'.format(' | '.join(violations)))
+
+@help_file_entry_rule
+def mismatched_help_parameter_rule(linter, help_entry):
+    if help_entry not in linter.commands:
+        return
+
+    help_names = linter.get_help_entry_example_names(help_entry)
+    help_texts = linter.get_help_entry_example_text(help_entry)
+    violations = []
+    for help_name, help_text in zip(help_names, help_texts):
+        if help_name not in help_text:
```

Comment:
example names are lengthy descriptions of what an example is is doing and will likely not appear in the example text
Please test your new linter rule thoroughly.

Reply:
Ditto what @williexu said.

Essentially the logic you are shooting for (in pseudocode) is:
```Python
# detect mismatch between example and supported options
for option in example_text:
  if option not in command.arguments.option_list:
    violations.append(option)
```

This will detect when an option or flag in the example text isn't actually supported by the command.  Then there's also this:

```Python
# detect when example is for wrong command
if not example_text.startswith(help_entry):
   violations.append(example_text)
```


This could catch copy-paste errors where an example from one command is pasted to another. However, you will run into problems for certain multi-step examples.